src/kraken/std/cargo/__init__.py

```diff
@@ -267,6 +273,9 @@ def cargo_publish(
         env=Supplier.of_callable(lambda: {**cargo.build_env, **(env or {})}),
     )
 
-    task.add_relationship(f":{CARGO_BUILD_SUPPORT_GROUP_NAME}?")
+    task.add_relationship(f":{CARGO_BUILD_ESSENTIAL_GROUP_NAME}?")
+
+    if auth_proxy:
```

Comment:
I still like the idea of making this based on `verify` because if `cargo publish` wants to verify the package, it has to build it, and any cargo command that _builds_ should depend on the build support group.

```suggestion
    if verify:
```